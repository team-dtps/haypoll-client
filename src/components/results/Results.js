import React from 'react';
import PropTypes from 'prop-types';
import Choice from './Choice';

function Results({ results }) {
  const { prompt, voteTotals } = results;
  const resultsList = voteTotals.map((option, i) => {
    return <Choice key={i} total={option.total} choice={option.choice} />;
  });
  return (
    <section>
      <h1>{prompt}</h1>
      <ul>
        {resultsList}
      </ul>
    </section>
  );
}

Results.propTypes = {
  results: PropTypes.object.isRequired
};

export default Results;
