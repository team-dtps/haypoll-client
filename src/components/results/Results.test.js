import React from 'react';
import renderer from 'react-test-renderer';
import Results from './Results';

describe('Results snapshot', () => {
  it('can match a snapshot', () => {
    const props = {
      results: {
        prompt: 'do you like hay?',
        voteTotals: [
          { choice: 'yay', total: 3 },
          { choice: 'nay', total: 5 }
        ]
      }
    };

    const tree = renderer.create(
      <Results results={props.results}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
