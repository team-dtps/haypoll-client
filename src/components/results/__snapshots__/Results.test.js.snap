// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results snapshot can match a snapshot 1`] = `
<section>
  <h1>
    do you like hay?
  </h1>
  <ul>
    <li>
      yay
       = 
      3
      %
    </li>
    <li>
      nay
       = 
      5
      %
    </li>
  </ul>
</section>
`;
